### ISSUE TITLE
Type of `x`, `y` and `z` values of a Point are now `numpy.float64`

### ISSUE THREAD

**Issue by jordibeen** [shapely/shapely#2070]
With NumPy's 2.0 release, they've [changed the representation of their scalars](https://numpy.org/devdocs/release/2.0.0-notes.html#representation-of-numpy-scalars-changed). This has caused the `x`, `y`, and `z` attribute of a Point to no longe be [Python's built-in float type](https://docs.python.org/3/library/functions.html#float), but [NumPy's float64 type](https://numpy.org/doc/stable/reference/arrays.scalars.html#numpy.float64) instead.

```sh
>>> from shapely import Point
>>> point = Point(1, 2)
>>> type(point.x)
<class 'numpy.float64'>
```

However, according to Shapely's documentation of [Point](https://shapely.readthedocs.io/en/stable/reference/shapely.Point.html#shapely.Point), these attributes should be the normal `float` type instead. 

---

This change caused issues in our project because we were passing a Point's `x` and `y` value to our ORM (SQLAlchemy), which was unable to convert a `numpy.float64` to a database value.

What is the plan for Shapely going forward? I think it would make sense to have Shapely return Python built-in types, but if you're planning to take the NumPy approach, then the documentation should be updated.

**Comment by mwtoews** [comment]
Agree this is a minor nuisance, it would be nice to have this fixed for the next minor release of the 2.0.x series.

I'll need to investigate a bit deeper, as there are subtle nuances between some properties and top-level functions, e.g.:
```pycon
>>> import shapely
>>> point = shapely.from_wkt("point(1 2)")
>>> poly = point.buffer(1.0)
>>> shapely.area(poly)
np.float64(3.136548490545939)
>>> poly.area
3.136548490545939
```
ideally all scalar numbers would be `float` built-in types.

**Comment by jorisvandenbossche** [comment]
> This has caused the `x`, `y`, and `z` attribute of a Point to no longe be [Python's built-in float type](https://docs.python.org/3/library/functions.html#float), but [NumPy's float64 type](https://numpy.org/doc/stable/reference/arrays.scalars.html#numpy.float64) instead.

FWIW, this did not actually change: also with NumPy 1.26, the return value was a `numpy.float64` scalar. But as you mentioned, the `repr` of the numpy scalars changed in NumPy 2.0.0, making it now much more obvious that the return value is indeed a numpy scalar and not a Python float. 
So I am not entirely sure why other libraries like SQLAlchemy started to fail on numpy scalars (maybe there was another change to how scalars behave in NumPy 2.0), but so we already have been returning such numpy scalars in 2.0.x for more than a year.

That said, as mentioned above, we should certainly make this consistent and have all geometry object attributes/methods return python scalars where possible.
